src/vmprof_main.h

```diff
@@ -207,6 +243,17 @@ static void sigprof_handler(int sig_nr, siginfo_t* info, void *ucontext)
     __sync_lock_release(&spinlock);
 #endif
 
+#ifdef VMPROF_LINUX
+    // SIGNAL ABUSE AHEAD
+    // On linux, the prof timer will deliver the signal to the thread which triggered the timer,
+    // because these timers are based on process and system time, and as such, are thread-aware.
+    // For the real timer, the signal gets delivered to the main thread, seemingly always.
+    // Consequently if we want to sample all threads, we need to forward the signal with tgkill.
+    if ((signal_type == SIGALRM) && (pid == tid)) {
+        broadcast_signal_for_threads(pid);
```

Comment:
I'm pretty sure these signals are allowed to be coalesced / swallowed, yes.
I don't think there's a good way to detect this, but I might be wrong?

Possible workarounds:
1. Ignore it, when we read the profile later we can check for lost signals / stack traces if we know the threads were present for the whole profile duration. If we don't know that, it gets tricky.
2. Trigger a real-time signal from the main thread instead, and set other threads to catch that real-time signal instead of the SIGALRM from the timer?
3. Other?